cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -610,15 +605,10 @@ __global__ void computeSplitRegressionKernel(
   // transfer from global to smem
```

Comment:
`gs.sync()` on line 603 is not required anymore as there is no second pass. This change plus the single block bypass trick I implemented as part of #3818 performed 10-12% better in my testing.